src/storage-controller/src/lib.rs

```diff
@@ -1851,11 +1851,6 @@ where
                             .read_hold
                             .try_downgrade(Antichain::new())
                             .expect("must be possible");
-                    } else {
-                        soft_panic_or_log!(
-                            "DroppedIds for ID {id} but we have neither ingestion nor export \
-                             under that ID"
-                        );
```

Comment:
Would be good to have a comment here explaining when this can happen and why it's fine to ignore it.

I assume in a multi-replica cluster all replicas send a `DroppedIds` and after the first one we have removed the `self.collections` entry so we end up here? I'm not quite understanding what the intention behind `DroppedIds` is though: I thought it was so the controller would know to keep around resources that are still used by replicas, but with this change it would do so only for the fastest replica.

Reply:
After what we talked about today I added some code with the simplest way I would think of that handles multiple `DroppedIds` responses.